### PR TITLE
refactor: remove npm snyk test path, use nodejs-plugin

### PR DESCRIPTION
**Pull request by orsagie**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Two parts to this refactor:
* merge the `npm` path in `snyk-test` to the main path
* replace the npm plugin in  the `lib/plugins` with the test one. 

Next step is to pull the npm-plugin to its own repo and use the nodejs parser.

##### Where to start reviewing

src/lib/snyk-test/index.js

##### Tickets

https://snyksec.atlassian.net/browse/BST-582

### CODE REVIEW

File: src/lib/plugins/index.ts [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
 import * as rubygemsPlugin from './rubygems';
 import * as mvnPlugin from 'snyk-mvn-plugin';
 import * as gradlePlugin from 'snyk-gradle-plugin';
 import * as sbtPlugin from 'snyk-sbt-plugin';
-import * as yarnPlugin from './yarn';
 import * as pythonPlugin from 'snyk-python-plugin';
 import * as goPlugin from 'snyk-go-plugin';
 import * as nugetPlugin from 'snyk-nuget-plugin';
 import * as phpPlugin from 'snyk-php-plugin';
+import * as nodejsPlugin from './npm-plugin';
```
Comment (kyegupov) [inline src/lib/plugins/index.ts]: I'd rather rename the directory as well.
Imports that are changing name are confusing.
正名 !

File: src/lib/plugins/npm-plugin/index.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import * as modulesParser from './npm-modules-parser';
+import * as lockParser from './npm-lock-parser';
+import * as types from '../types';
+
+export async function inspect(root: string, targetFile: string, options: types.Options = {}):
+Promise<types.InspectResult> {
+  const isLockFileBased = (targetFile.endsWith('package-lock.json') || targetFile.endsWith('yarn.lock'));
+  if (targetFile.endsWith('yarn.lock') && getRuntimeVersion() < 6) {
```
Comment (kyegupov) [inline src/lib/plugins/npm-plugin/index.ts]: again, node < 6 is not a thing anymore, mwahaha.

File: src/lib/snyk-test/run-test.ts [hunk 3]
```diff
@@ -32,26 +32,28 @@ interface Payload {
     authorization: string;
   };
   body?: {
-    depGraph: depGraphLib.DepGraph,
+    depGraph?: depGraphLib.DepGraph,
```
Comment (kyegupov) [inline src/lib/snyk-test/run-test.ts]: when would it be missing?

**Comment by orsagie** [comment]
@kyegupov This PR is on top of the previous one. You can just review the latest commit https://github.com/snyk/snyk/pull/483/commits/a3f221acfeb9cc2e7181afc2f7e3ef0de7d79462

File: src/lib/snyk-test/run-test.ts (hunk 3 above)
Comment (orsagie) [reply src/lib/snyk-test/run-test.ts]: if vulnEndpoint is specified. This is an old option and we may be able to remove it, but I did not want to change the actual behavior of the CLI.

File: src/lib/plugins/index.ts (hunk 1 above)
Comment (orsagie) [reply src/lib/plugins/index.ts]: わかりました

**Comment by lili2311** [comment]
Please manually verify this PR for the following scenarios:
1. Npm with lockfile
- test works the same as before (use latest cli version for comparison)
- wizard works the same as before (use latest cli version for comparison)
- patch works the same as before (use latest cli version for comparison)

2. Yarn lockfile based
- test works the same as before (use latest cli version for comparison)
- wizard works the same as before (use latest cli version for comparison) => pick some issues to patch + some to fix. all should succeed
- patch works the same as before (use latest cli version for comparison)

3. Npm (npm < 5, no lockfile package.json only)
- test works the same as before (use latest cli version for comparison)
- wizard works the same as before (use latest cli version for comparison)
- patch works the same as before (use latest cli version for comparison)
- when no `node_modules` are present we show a relevant error asking the user to run `npm i`

**Comment by kyegupov** [comment]
@lili2311 tested, no problems since the yarn issue was fixed.

**Pull request merged**